Create AdminUser class to inherit from User super class

nzediegwu1 opened the pull request:
*redefine sendNotification method to send admin a notification when user deletes own even for a particular center: to demonstrate polymorphism
*create object of AdminUser class and call createEvent inherited method to demonstrate inheritance

Commit:
Create AdminUser class to inherit from User super class

*redefine sendNotification method to send admin a notification when user deletes own even for a particular center: to demonstrate polymorphism
*create object of AdminUser class and call createEvent inherited method to demonstrate inheritance

Commit:
Define createCenter method in AdminUser class

define validateCenter method in AdminUser
create centers global array with dummy data
call createCenter method from adminUser object

Commit:
remove unnecessary white spaces

Commit:
include pull -request template

oreoluwade reviewed (commented):
- This is quite a good effort, but I don't see an indication (comment) anywhere of the class showing evidence of polymorphism.

- Is your project a node project? If no, is there a reason why?

nzediegwu1 commented:
Polymorphism was implemented in line 97 in the first commit, using sendNotification method in AminUser class which overrides the sendnotification method in user class at line 60

nzediegwu1 commented:
The project is not a node project, but rather a javascript console program. Since test was not included, I thought there was no need creating a package.json for running test script

oreoluwade commented:
Okay. Your explanation for why it is not a node project is acceptable. However, I would like to say that it is not only for the purpose of testing that a node project has to be initialized. In this project for example, if you made it a node project, and you specified a script in your `package.json` such as `"start": "node index.js"` (if the file that has your code is `index.js` for example), then I can easily initialize an object each time from the classes supplying the values the constructor defines. Then when I run `npm start` in the terminal, I can see the results of my initialization and method calls as it were. 

If you need more clarification on this, let me know.

As far as where polymorphism was implemented is concerned, I prefer that you make a comment on the line where it was implemented and make a commit to this same branch just as you did for inheritance and co instead of writing it only as a relply to my comment on the PR.

nzediegwu1 commented:
Thanks a lot for the explanation and insight, its well understood. I would add proper comment for the polymorphism implementation and make it a node project shortly

Commit:
Include package.json and response to feedback

*rearrange code for readability
*add comment for polymorphism implementation
*improve overall comment quality

oreoluwade commented on app.js:
```diff
@@ -0,0 +1,145 @@
+const newCenter = adminUser.createCenter('National Stadium', 'Abuja', 55000);
+const newAdminnotification = adminUser.sendNotification('Hackathon', 'Andela Tower');
+
+//display result
```
Excellent!

nzediegwu1 replied:
Thanks a lot for the commendation, I'm inspired

oreoluwade commented on package.json:
```diff
@@ -0,0 +1,19 @@
+  "version": "1.0.0",
+  "description": "An OOP implementation that models a real-world scenario/problem in an Event Management application, allowing user class to create events and send notification, while admin user class, which inherits from the user class, can create a center, create an event by inheritance and send notification by method overriding",
+  "scripts": {
+    "start": "node app.js"
```
I love your swift and effective response to feedback. Keep it up.

nzediegwu1 replied:
Thank you

oreoluwade reviewed (approved):
You are free to merge this now. Good work!

nzediegwu1 committed:
Merge branch 'develop' into ft-admin-user-class
